crawl4ai/async_crawler_strategy.py

```
@@ -900,12 +901,43 @@ async def handle_request_failed_capture(request):
+                viewport = page.viewport_size
+                if viewport is None:
+                    # Set default viewport if not already set
+                    await page.set_viewport_size(
+                        {"width": self.browser_config.viewport_width, "height": self.browser_config.viewport_height}
+                    )
+                    viewport = page.viewport_size
+
+                # Ensure viewport is valid before proceeding
+                if viewport and viewport.get("width") and viewport.get("height"):
+                    # Generate random start and end points for mouse movement
+                    start_x = random.randint(50, viewport["width"] // 3)
```

Ruff (0.14.0)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)